lib/actions/bulk_cud.js

```diff
@@ -20,27 +21,26 @@ exports.objectTypes = async function objectTypes(configuration) {
   }
 };
 
-exports.process = async function bulkCUD(message, configuration) {
+module.exports.process = async function bulkCUD(message, configuration) {
   this.logger.info('Starting Bulk %s action', configuration.operation);
   // Get CSV from attachment
-  if (!message.attachments || Object.keys(message.attachments).length === 0) {
-    this.logger.error('Attachment not found');
-    return messages.newMessageWithBody({ });
-  }
-
-  const key = Object.keys(message.attachments)[0];
 
   let timeout = parseInt(configuration.timeout, 10);
   if (Number.isNaN(timeout) || (timeout < 0)) {
     this.logger.warn(`Timeout is incorrect. Set default value: ${DEFAULT_TIMEOUT} sec.`);
     timeout = DEFAULT_TIMEOUT;
   }
+  let { attachmentUrl } = message.body;
+  if (!attachmentUrl) {
+    if (!message.attachments || Object.keys(message.attachments).length === 0 || !message.attachments[Object.keys(message.attachments)[0]].url) {
```

Comment:
Can't this be simplified to 
!message.attachments[0].url?

Reply:
`message.attachments` is an object where file name is a key and this object mast have `url` key inside

Reply:
Ok. Got it. I was thinking of the wrong message.attachment structure. Then it's fine. But I'd better increase readability moving it into a method with a self-explanatory name.